src/hooks/useBreakpoint.tsx

```diff
@@ -0,0 +1,41 @@
+import equal from "fast-deep-equal";
+import { useEffect, useState } from "react";
+import { useDebouncedCallback } from "use-debounce";
+import { BreakpointKey, Breakpoints } from "../Css";
+
+/**
+ * A React hook to return a record of responsive breakpoints that updates on resize.
+ *
+ * @example
+ * const { breakpoints } = useBreakpoint();
+ * if(breakpoints.mdAndDown) {...do something cool}
+ */
+
+type BreakpointsType = Record<BreakpointKey, boolean>;
+
+export const useBreakpoint = (): BreakpointsType => {
+  const [breakpoints, setBreakpoints] = useState(matchMediaBreakpoints());
+
+  const handleResize = useDebouncedCallback(() => {
+    const newBps = matchMediaBreakpoints();
+    if (equal(breakpoints, newBps)) return;
+
+    setBreakpoints(newBps);
+  }, 1000);
```

Comment:
**nit:** 1 second seems like a fairly large delay. Feels like it can be a bit sluggish to respond. How did you land on using 1 second?